src/main/java/org/trigon/hardware/SignalThreadBase.java

```diff
@@ -7,21 +7,37 @@
 import java.util.concurrent.ArrayBlockingQueue;
 import java.util.concurrent.locks.ReentrantLock;
 
+/**
+ * A class that represents a base for a signal thread.
+ */
 public class SignalThreadBase extends Thread {
```

Comment:
explain more about the signal thread.
the implementation of this thread also need some javadocs. 
I believe there are also some fixes need to be done from the debugging branch.

Reply:
the debugging branch doesn't exist

Reply:
Weird. When was it deleted?

Reply:
idk, I didn't

Reply:
luckily the branch was saved on my computer, https://github.com/Programming-TRIGON/TRIGONLib/compare/main...debugging